.github/actions/update-axe-core-v1/src/getPackageManager.ts

```diff
@@ -0,0 +1,26 @@
+import { promises as fs } from 'fs'
+import path from 'path'
+import type { getPackageManagerReturn } from './types'
+
+export default async function getPackageManager(
+  dirPath: string
+): getPackageManagerReturn {
+  if (await exists(path.join(dirPath, 'package-lock.json'))) {
+    return 'npm'
+  }
+
+  if (await exists(path.join(dirPath, 'yarn.lock'))) {
+    return 'yarn'
+  }
+
+  return undefined
+}
+
+async function exists(filePath: string): Promise<boolean> {
+  try {
+    await fs.stat(filePath);
```

Comment:
Asynchronously checking if something exists doesn't make sense. It may exist while this function began and could be gone by the time it's done. This is exactly why Node.js removed `fs.exists()`.

Reply:
Node deprecated `fs.exists` because it [didn't follow node's return style](https://nodejs.org/api/fs.html#fsexistssyncpath).

> fs.exists() is deprecated, but fs.existsSync() is not. The callback parameter to fs.exists() accepts parameters that are inconsistent with other Node.js callbacks. fs.existsSync() does not use a callback.

[`fs.exists` docs says to use `fs.stat` instead](https://nodejs.org/api/fs.html#fsexistspath-callback) for this reason.

> Stability: 0 - Deprecated: Use fs.stat() or fs.access() instead.

Reply:
> It may exist while this function began and could be gone by the time it's done.

This is mostly orthogonal to the choice of async vs sync exists function; this is something you need to consider anytime you do separate calls for exists vs use of a file, even if you use a sync exist check.